modules/core/src/lifecycle/create-props.js

```diff
@@ -2,40 +2,46 @@ import {applyPropOverrides} from '../lib/seer-integration';
 import log from '../utils/log';
 import {parsePropTypes} from './prop-types';
 
-export const EMPTY_ARRAY = Object.freeze([]);
-
 // Create a property object
 export function createProps() {
   const component = this; // eslint-disable-line
 
   // Get default prop object (a prototype chain for now)
-  const {defaultProps} = getPropsPrototypeAndTypes(component.constructor);
+  const propsPrototype = getPropsPrototypeAndTypes(component.constructor).defaultProps;
 
-  // Create a new prop object with  default props object in prototype chain
-  const newProps = Object.create(defaultProps, {
+  // Create a new prop object with default props object in prototype chain
+  const propsInstance = Object.create(propsPrototype, {
+    // Props need a back pointer to the owning component
     _component: {
       enumerable: false,
       value: component
     },
-    _asyncProps: {
+    // The supplied (original) values for those async props that are set to url strings or Promises.
+    // In this case, the actual (i.e. resolved) values are looked up from component.internalState
+    _asyncPropOriginalValues: {
```

Comment:
What is the thought behind having multiple `_asyncProp*Values` objects? Does it make more sense to use a single `_asyncProps` each in the shape of `{originalValue, resolvedValue}`?

Reply:
Looking at it again: the reason for the flat structure is that these two objects are added the prop object **instance** (i.e. not the prototype). We presumably want to avoid creating a bunch of small objects every time a layer is instantiated.